### PR TITLE
Better Errors for fs API

### PR DESCRIPTION
**Pull request by Wilma456**
The Errors of the fs API now contain the Path.

### CODE REVIEW

**Comment by SquidDev** [comment]
I'm not sure this current syntax reads great, maybe a couple of alternatives:

 - `'foo' is not a directory`. This doesn't work that well for other error messages like `Access denied` though.
 - `Not a directory (when listing 'foo')` - this works for everything, but is rather verbose
 - `foo: not a directory` - This is what Lua does on my machine, again it works with everything.

**Comment by dan200** [comment]
Agree with SquidDev: including the path is good, but need better strings

**Comment by lupus590** [comment]
Error string suggestion: `received fs path foo: not a directory`

**Comment by Wilma456** [comment]
@dan200 I had updated the messages to a new style

fserror

**Comment by lupus590** [comment]
The new format is similar to Vanilla Lua while still being clear that `/foo` is not a part of the file which errored.

File: src/main/java/dan200/computercraft/core/filesystem/FileSystem.java [hunk 1]
```diff
@@ -730,7 +730,7 @@ private MountWrapper getMount( String path ) throws FileSystemException
         }
         if( match == null )
         {
-            throw new FileSystemException( "Invalid Path" );
+            throw new FileSystemException( "Invalid Path (" + path + ")" );
```
Comment (Lignum) [inline src/main/java/dan200/computercraft/core/filesystem/FileSystem.java]: Shouldn't this be `"/" + path + ": Invalid path"` to be consistent with the others?

**Comment by Lignum** [comment]
Don't repeat yourself; that should probably be a function:
```Java
static String formatFSError( String path, String message )
{
    return String.format( "/%s: %s", path, message );
}
```
It's cleaner code, and that'll let us switch the format more easily if needed.

**Comment by Wojbie** [comment]
Is this really such big thing that it should require its own function?

**Comment by Lignum** [comment]
@Wojbie I suppose it's a matter of taste. Coming from functional programming, I'm a fan of generalising as much as possible.

File: src/main/java/dan200/computercraft/core/filesystem/FileSystem.java (hunk 1 above)
Comment (Wilma456) [inline src/main/java/dan200/computercraft/core/filesystem/FileSystem.java]: Forget it, sorry

**Comment by Wilma456** [comment]
I think, dan200 should decide, if we add a function for this.

**Pull request merged**